File: src/main/java/eu/karelhovorka/zpevnik/util/SongSettingsFormatter.java
package eu.karelhovorka.zpevnik.util;

import eu.karelhovorka.zpevnik.text.SongDisplaySettings;

import static eu.karelhovorka.zpevnik.util.Preconditions.checkNotNull;


public class SongSettingsFormatter {


    public static String modifyTextBasedOnSettings(String source, SongDisplaySettings settings) {
        checkNotNull(source, "source");
        checkNotNull(settings, "settings");
        String text = source;
        text = ChordDetector.replacePlainChords(text);
        if (!settings.isDisplayText()) {
            text = Transposer.removeNonChords(text);
        }
        if (settings.isHideIdenticalSequences()) {
            text = Transposer.removeDuplicateChordSequences(text);
        }
        if (settings.isDisplayChords()) {
            text = ToneTransposer.transposeAll(text, settings.getInterval(), settings.getCountryCategory(), settings.getModificationAbbreviation());
        } else {
            text = Transposer.removeChords(text);
        }
        return StringUtil.removeTrailingWhitespace(text);
    }
}

File: src/main/kotlin/eu/karelhovorka/zpevnik/util/SongSettingsFormatter.kt
package eu.karelhovorka.zpevnik.util

import eu.karelhovorka.zpevnik.text.SongDisplaySettings
import mock.JvmStatic


object SongSettingsFormatter {

    @JvmStatic
    fun modifyTextBasedOnSettings(source: String, settings: SongDisplaySettings = SongDisplaySettings.DEFAULT): String {
        var text = source
        text = ChordDetector.replacePlainChords(text)
        if (!settings.isDisplayText) {
            text = Transposer.removeNonChords(text)
        }
        if (settings.isHideIdenticalSequences) {
            text = Transposer.removeDuplicateChordSequences(text)
        }
        if (settings.isDisplayChords) {
            text = ToneTransposer.transposeAll(text, settings.interval, settings.countryCategory, settings.modificationAbbreviation)
        } else {
            text = Transposer.removeChords(text)
        }
        return text.trimEnd()
    }
}